CMakeLists.txt

```diff
@@ -0,0 +1,173 @@
+
+option(WANNIER90_MPI "Wannier90: Build with MPI support" OFF)
+option(WANNIER90_SHARED_LIBS "Wannier90: Build library as a shared library" ${PROJECT_IS_TOP_LEVEL})
+option(WANNIER90_INSTALL "Wannier90: Install files" ${PROJECT_IS_TOP_LEVEL})
+option(WANNIER90_TEST "Wannier90: Build test-suite" ${PROJECT_IS_TOP_LEVEL})
+
+#[=============================================================================[
+#                            Project configuration                            #
+]=============================================================================]
+
+if (NOT CMAKE_Fortran_MODULE_DIRECTORY)
+	set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/fortran_mods)
+endif ()
+set(BUILD_SHARED_LIBS ${WANNIER90_SHARED_LIBS})
+
+if (WANNIER90_INSTALL)
+	include(GNUInstallDirs)
+	include(CMakePackageConfigHelpers)
+
+	# CMake does not properly support fortran module installation paths.
+	# https://gitlab.kitware.com/cmake/cmake/-/issues/19608
+	# https://discourse.cmake.org/t/api-design-c-modules-source-listings-and-interface-properties/5389/14
+	# Note: This does not follow fortran-stdlib format, instead it follows Fedora guidelines
+	# https://pagure.io/packaging-committee/issue/1334
+	cmake_path(APPEND CMAKE_INSTALL_LIBDIR "${CMAKE_Fortran_COMPILER_ID}-${CMAKE_Fortran_COMPILER_VERSION}" modules
+			OUTPUT_VARIABLE _DEFAULT_CMAKE_INSTALL_MODULEDIR
+	)
+	set(CMAKE_INSTALL_MODULEDIR ${_DEFAULT_CMAKE_INSTALL_MODULEDIR}
+			CACHE STRING
+			"Fortran module installation path (Not a cmake native variable)"
+	)
+	cmake_path(IS_ABSOLUTE CMAKE_INSTALL_MODULEDIR _is_absolute)
+	if (_is_absolute)
+		set(CMAKE_INSTALL_FULL_MODULEDIR ${CMAKE_INSTALL_MODULEDIR})
+	else ()
+		cmake_path(APPEND CMAKE_INSTALL_PREFIX ${CMAKE_INSTALL_MODULEDIR}
+				OUTPUT_VARIABLE CMAKE_INSTALL_FULL_MODULEDIR
+		)
+	endif ()
+endif ()
+
+## Populate variables for FetchContent and sub-projects
+# All variables have to be consistent with CMakeExtraUtilsConfig.cmake
+set(Wannier90_MPI ${WANNIER90_WITH_MPI})
+
+#[=============================================================================[
+#                              External packages                              #
+]=============================================================================]
+
+## Third-party libraries
+find_package(BLAS REQUIRED)
+find_package(LAPACK REQUIRED)
+if (WANNIER90_MPI)
+	find_package(MPI REQUIRED)
+endif ()
+
+#[=============================================================================[
+#                               Main definition                               #
+]=============================================================================]
+
+## Main targets
+add_executable(Wannier90_exe)
+set_target_properties(Wannier90_exe PROPERTIES
+		OUTPUT_NAME wannier90.x
+		EXPORT_NAME exe
+)
+add_executable(Wannier90::exe ALIAS Wannier90_exe)
+add_executable(Wannier90_post)
+set_target_properties(Wannier90_post PROPERTIES
+		OUTPUT_NAME postw90.x
+		EXPORT_NAME post
```

Comment:
Could you please remove `.x` from executable name and set `CMAKE_EXECUTABLE_SUFFIX='.x'` on `UNIX` instead.

On Windows/MinGW-w64 the executables are wannier90.x.exe and postw90.x.exe

Reply:
Good catch 👍